frontend/src/pages/profile/components/avatar-wrapper/avatar-wrapper.tsx

```diff
@@ -0,0 +1,103 @@
+import { type ChangeEvent, type FC } from 'react';
+
+import { Avatar, Button, ErrorMessage } from '~/libs/components/components.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback, useRef, useState } from '~/libs/hooks/hooks.js';
+import { fileApi } from '~/libs/packages/image/file-api.js';
+import { type UserAuthResponseDto } from '~/packages/users/users.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  user: UserAuthResponseDto;
+  errorImageUpload: string | null;
+  className?: string;
+  onRemoveAvatar: () => void;
+  onUpdateAvatarId: (value: number | null) => void;
+  onErrorUpload: (value: string | null) => void;
+};
+
+const AvatarWrapper: FC<Properties> = ({
+  user,
+  errorImageUpload,
+  onRemoveAvatar,
+  onUpdateAvatarId,
+  onErrorUpload,
+}): JSX.Element => {
+  const [previewUrl, setPreviewUrl] = useState<string | null>(user.avatarUrl);
+  const reference = useRef<HTMLLabelElement>(null);
+
+  const handleUploadUserAvatar = useCallback(
+    (event_: ChangeEvent<HTMLInputElement>): void => {
+      const [image] = event_.target.files ?? [];
+
+      if (image) {
+        const formData = new FormData();
+        formData.append('file', image);
+
+        void fileApi
+          .uploadFile(formData)
+          .then((fileData) => {
+            onUpdateAvatarId(fileData.id);
+            setPreviewUrl(fileData.url);
+            onErrorUpload(null);
+          })
+          .catch((error: unknown) => {
+            if (error instanceof Error) {
+              setPreviewUrl(null);
+              onErrorUpload(error.message);
+            }
+          });
+      }
+    },
+    [onErrorUpload, setPreviewUrl, onUpdateAvatarId],
+  );
+
+  const handleRemoveAvatar = useCallback(() => {
+    onRemoveAvatar();
+    setPreviewUrl(null);
+    onErrorUpload(null);
+  }, [onErrorUpload, onRemoveAvatar]);
+
+  const handleUpdateAvatar = useCallback(() => {
+    if (!reference.current) {
+      return null;
+    }
+
+    reference.current.click();
+  }, []);
+
+  return (
+    <div className={styles.imageGroup}>
+      <label className={styles.imageWrapper} ref={reference} htmlFor="avatarId">
+        <Avatar
+          username={`${user.firstName} ${user.lastName}`}
+          avatarUrl={previewUrl}
+          className={styles.avatar}
+        />
+        <input
+          className="visually-hidden"
+          id="avatarId"
+          onChange={handleUploadUserAvatar}
+          type="file"
+          accept=".jpg, .jpeg, .png, .webp, .bmp, .svg"
+        />
+      </label>
+      <div className={styles.imageGroupControls}>
+        <Button
+          className={getValidClassNames(styles.control, styles.imageUpdate)}
+          onClick={handleUpdateAvatar}
+          label="Update"
+        />
```

Comment:
I believe we don't need separate button for update - as file input is already an updating field
![image](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/assets/19575839/5cf9466c-2506-4f2b-adcb-46fc5fb2845c)
Just add a lighter or darker (what would look better) background with edit icon on hovering field with avatar - to show that it's editable

Reply:
done